superduperdb/backends/ibis/query.py

```diff
@@ -49,25 +51,55 @@ def _model_update_impl(
     outputs: t.Sequence[t.Any],
     flatten: bool = False,
 ):
-    if flatten:
-        raise NotImplementedError('Flatten not yet supported for ibis')
+    if CFG.cluster.vector_search.type == 'in_memory':
+        if flatten:
+            raise NotImplementedError('Flatten not yet supported for ibis')
 
-    if not outputs:
-        return
-
-    table_records = []
-    for ix in range(len(outputs)):
-        d = {
-            '_input_id': str(ids[ix]),
-            'output': outputs[ix],
-        }
-        table_records.append(d)
+        if not outputs:
+            return
 
-    for r in table_records:
-        if isinstance(r['output'], dict) and '_content' in r['output']:
-            r['output'] = r['output']['_content']['bytes']
+        table_records = []
+        for ix in range(len(outputs)):
+            d = {
+                '_input_id': str(ids[ix]),
+                'output': outputs[ix],
+            }
+            table_records.append(d)
+
+        for r in table_records:
+            if isinstance(r['output'], dict) and '_content' in r['output']:
+                r['output'] = r['output']['_content']['bytes']
+
+        db.databackend.insert(f'_outputs.{predict_id}', table_records)
+
+    elif CFG.cluster.vector_search.type == 'pg_vector':
+        # Connect to your PostgreSQL database
+        conn = psycopg2.connect(CFG.cluster.vector_search.uri)
+        table_name = f'_outputs.{predict_id}'
+        with conn.cursor() as cursor:
+            cursor.execute('CREATE EXTENSION IF NOT EXISTS vector')
+            cursor.execute(f"""DROP TABLE IF EXISTS "{table_name}";""")
+            cursor.execute(
+                f"""CREATE TABLE "{table_name}" (
+                    _input_id VARCHAR PRIMARY KEY,
+                    output vector(1024),
+                    _fold VARCHAR
+                );
+                """
+            )
+            for ix in range(len(outputs)):
+                try:
+                    cursor.execute(
+                        f"""INSERT INTO "{table_name}" (_input_id, output) VALUES (%s, %s);""",
+                        [str(ids[ix]), outputs[ix]]
+                    )
+                except:
+                    pass
 
-    db.databackend.insert(f'_outputs.{predict_id}', table_records)
+        # Commit the transaction
+        conn.commit()
+        # Close the connection
+        conn.close()
```

Comment:
We should load the vectors into a specific table only when initializing vector search for the first time, otherwise we will not be able to switch the engine of vector search.

So we don't need to add vectors into the table here.

WDYT? @kartik4949

Reply:
Don't worry about duplication, we have `_input_id` as a PRIMARY KEY while creating vector table `_outputs.listener1::0`. I have checked there is no duplication.